torchx/test/torchx/nx_test.exs

```diff
@@ -148,6 +148,16 @@ defmodule Torchx.NxTest do
                Nx.tensor([[0, 1, 2], [3, 4, 5]], backend: Nx.BinaryBackend)
     end
 
+    test "make_diagonal" do
+      t =
+        [1, 2, 3]
+        |> Nx.tensor()
+        |> Nx.make_diagonal()
+
+      assert Nx.backend_transfer(t) ==
+               Nx.tensor([[1, 0, 0], [0, 2, 0], [0, 0, 3]], backend: Nx.BinaryBackend)
```

Comment:
You cam use assert_all_close to avoid the need to backend_transfer

Reply:
> You cam use assert_all_close to avoid the need to backend_transfer

@polvalente quick question: Why aren't we using assert_all_close in the `Nx.iota` test?

https://github.com/elixir-nx/nx/blob/506ac004efdd1076e033b784db752d6996f012da/torchx/test/torchx/nx_test.exs#L144-L149

Reply:
It's probably one of the tests that came before the assertion? 

Although thinking about it we should actually assert exactly for when the tensor is expected to be exactly equal to the other...

I'll send a PR with a new "assert_*" function for this

Reply:
> Although thinking about it we should actually assert exactly for when the tensor is expected to be exactly equal to the other...

In that case should `iota` and `make_diagonal` use the `assert_*` function?

Reply:
Yes. I'm thinking about calling it assert_equal. 

It would be something like assert_all_close but using left |> Nx.equal(right) |> Nx.all() |> Nx.to_number() == 1 as the equality check.

Do you want to send a PR on this or should I?

Reply:
I think I'll send one.

Reply:
Great!